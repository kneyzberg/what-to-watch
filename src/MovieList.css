.MovieList-container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: flex-start;
  align-items: center;
  gap: 30px;
  padding-left: 200px;
  padding-right: 200px;
}

@include media-breakpoint-up(lg) {
  .MovieList-shrink {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: flex-start;
    align-items: center;
    gap: 30px;
    padding-left: 400px;
    padding-right: 400px;
  }
}
