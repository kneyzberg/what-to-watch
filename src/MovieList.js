import React from "react";
import MovieCard from "./MovieCard";

function MovieList({ imgUrl, posterSize, movies }) {
  return (
    <div>
      {movies.map((m) => (
        <MovieCard
          key={m.id}
          movie={m}
          imgUrl={imgUrl}
          posterSize={posterSize}
        />
      ))}
    </div>
  );
}

export default MovieList;
