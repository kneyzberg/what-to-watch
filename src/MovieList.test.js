import MovieList from "./MovieList";
import React from "react";
import { render } from "@testing-library/react";

const movies = [
  {
    id: 1,
    title: "Testing Movie Card1",
    release_date: "2021-09-22",
    poster_path: "/test.jpg",
  },
  {
    id: 2,
    title: "Testing Movie Card2",
    release_date: "2021-09-22",
    poster_path: "/test.jpg",
  },
  {
    id: 3,
    title: "Testing Movie Card3",
    release_date: "2021-09-22",
    poster_path: "/test.jpg",
  },
];
const imgUrl = "testingmovies.com";
const posterSize = "w-3";

describe("MovieList test", () => {
  test("MovieList renders correctly", () => {
    render(
      <MovieList movies={movies} imgUrl={imgUrl} posterSize={posterSize} />
    );
  });

  test("matches snapshot", () => {
    const { container, getByAltText } = render(
      <MovieList movies={movies} imgUrl={imgUrl} posterSize={posterSize} />
    );
    expect(container).toMatchSnapshot();
    expect(getByAltText("Poster for Testing Movie Card1")).toBeInTheDocument();
  });
  test("Title is in the document", () => {
    const { getByText } = render(
      <MovieList movies={movies} imgUrl={imgUrl} posterSize={posterSize} />
    );
    expect(getByText("Testing Movie Card2")).toBeInTheDocument();
  });
});
