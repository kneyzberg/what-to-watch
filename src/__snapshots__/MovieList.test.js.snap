// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList test matches snapshot 1`] = `
<div>
  <div>
    <div>
      <div>
        Testing Movie Card1
         
        2021-09-22
      </div>
      <img
        alt="Poster for Testing Movie Card1"
        src="testingmovies.com/w-3//test.jpg"
      />
    </div>
    <div>
      <div>
        Testing Movie Card2
         
        2021-09-22
      </div>
      <img
        alt="Poster for Testing Movie Card2"
        src="testingmovies.com/w-3//test.jpg"
      />
    </div>
    <div>
      <div>
        Testing Movie Card3
         
        2021-09-22
      </div>
      <img
        alt="Poster for Testing Movie Card3"
        src="testingmovies.com/w-3//test.jpg"
      />
    </div>
  </div>
</div>
`;
